CommonMark/Parser/InlineMethods.cs

```diff
@@ -953,7 +955,14 @@ static Reference ParseLinkDetails(Subject subj)
 
             // rollback the subject position because didn't match anything.
             subj.Position = endlabel;
-            return null;
+            if (supportPlaceholderBrackets)
+            {
+                return new Reference() { IsPlaceholder = true };
```

Comment:
this method should return the Url within the Reference instead of it being looked up at line 492